core/src/test/java/apoc/spatial/GeocodeTest.java

```diff
@@ -37,6 +41,38 @@ public void setUp() throws Exception {
         assumeRunningInCI();
     }
 
+    // -- with config map
+    @Test
+    public void testWrongUrlButViaOtherProvider() throws Exception {
+        // wrong url but doesn't fail because provider is osm, not opencage
+        testGeocodeWithThrottling("osm", false, 
+                map("url", "https://api.opencagedata.com/geocode/v1/json?q=PLACE&key=KEY111"));
+    }
+    
+    @Test(expected = QueryExecutionException.class)
+    public void testWrongUrlWithOpenCage() throws Exception {
+        // overwrite ApocConfig provider
+        testGeocodeWithThrottling("osm", false, 
+                map("provider", "opencage", "url", "https://api.opencagedata.com/geocode/v1/json?q=PLACE&key=KEY111"));
+    }
+    
+    @Test
+    public void testGeocodeOpenCageViaConfigMap() throws Exception {
+        Assume.assumeNotNull(OPENCAGE_KEY);
+        // overwrite ApocConfig provider
+        testGeocode("osm", 1000, false, 
+                map("provider", "opencage", "url", "https://api.opencagedata.com/geocode/v1/json?q=PLACE&key=KEY", "key", OPENCAGE_KEY));
+    }
+    
+    @Test
+    public void testReverseGeocodeOpenCageViaConfigMap() throws Exception {
+        Assume.assumeNotNull(OPENCAGE_KEY);
```

Comment:
These tests are always ignored in CI. How do we know they are doing what they are supposed to be doing? Can we mock or something?

Reply:
Also this particular test doesn't do anything with the key, and I don't know why. If this line gets commented out, the test passes

Reply:
I added/updated some tests in the existing `SpatialTest.java` which mocks the `spatial.*` procedures